pkg/server/api/bundle/v1/service.go

```diff
@@ -518,3 +569,50 @@ func applyBundleMask(b *types.Bundle, mask *types.BundleMask) {
 		b.JwtAuthorities = nil
 	}
 }
+
+func fieldsFromBundleProto(proto *types.Bundle, inputMask *types.BundleMask) logrus.Fields {
+	fields := logrus.Fields{
+		"trust_domain": proto.TrustDomain,
+	}
+
+	if inputMask == nil || inputMask.RefreshHint {
+		fields["refresh_hint"] = proto.RefreshHint
+	}
+
+	if inputMask == nil || inputMask.SequenceNumber {
+		fields["sequence_number"] = proto.SequenceNumber
+	}
+
+	if inputMask == nil || inputMask.JwtAuthorities {
+		for k, v := range fieldsFromJwtAuthoritiesProto(proto.JwtAuthorities) {
+			fields[k] = v
+		}
+	}
+
+	if inputMask == nil || inputMask.X509Authorities {
+		for k, v := range fieldsFromX509AuthoritiesProto(proto.X509Authorities) {
+			fields[k] = v
+		}
+	}
+	return fields
+}
+
+func fieldsFromJwtAuthoritiesProto(jwtAuthorities []*types.JWTKey) logrus.Fields {
+	fields := logrus.Fields{}
```

Comment:
```suggestion
	fields := make(logrus.Fields, 0, len(jwtAuthorities))
```

Reply:
it must be 3*len(), we add 3 per each jwtAuthority